### PR TITLE
Remove renovate, use dependabot for npm too

### LINKED ISSUE #20146: Please use Dependabot instead of Renovate
We want to use Dependabot consistently across all our (add-ons) repos. Please update the config to use Dependabot instead of Renovate. Thanks!

### PR DESCRIPTION
**Pull request by diox**
Fixes #20146

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 version: 2
 updates:
+- package-ecosystem: "npm"
+  directory: "/"
+  schedule:
+    interval: weekly
```
Comment (diox) [inline .github/dependabot.yml:6]: I figured that was good enough for addons-server

Comment (eviljeff) [reply .github/dependabot.yml:6]: generally, but for updates to our own packages like the linter we usually like it quicker

Comment (diox) [reply .github/dependabot.yml:6]: Good point

Comment (diox) [reply .github/dependabot.yml]: e1620da134273d95cbb8733014934ae604039452

**Review by eviljeff — APPROVED** [review]

**Pull request merged**